lib/server.js

```diff
@@ -164,6 +184,59 @@ module.exports = function createServer(config, log) {
   )
 
   api.post(
+    '/checkAuthenticated',
+    function (req, res, next) {
+      var action = req.body.action
+      var ip = req.body.ip
+      var uid = req.body.uid
+
+      if(!action || !ip || !uid){
+        var err = {code: 'MissingParameters', message: 'action, ip and uid are all required'}
+        log.error({op:'request.check', action: action, ip: ip, uid: uid, err: err})
+        res.send(400, err)
+        return next()
+      }
+
+      fetchUidRecord(uid)
+        .spread(
+          function (uidRecord) {
+
+            var retryAfter = uidRecord.addCount(action, uid)
+
+            return setUidRecord(uid, uidRecord)
```

Comment:
@larissagaulia we should also add support for `allowedEmailDomains`, means `@restmail.net` will not be limited.

```
if (allowedEmailDomains.isAllowed(email)) {
///
}
```

Ref: https://github.com/larissagaulia/fxa-customs-server/blob/87960655679a9fa909b2bdc325fc280765282144/lib/server.js#L151